doc/decisions/global_keyset.md

```diff
@@ -0,0 +1,37 @@
+# Global KeySet
+
+## Problem
+
+Some plugins need to communicate more data than is possible to do with metadata.
+This can limit the functionality of plugins, which need to exchange binary or very
+complex data.
+
+## Constraints
+
+## Assumptions
+
+## Considered Alternatives
+
+## Decision
+
+To make the communication between plugins easier, plugins will additionally
+get a handle to a global keyset. The global keyset is tied to a KDB handle,
+initialized on kdbOpen() and deleted on kdbClose().
+
+The global keyset handle is initialized and accessible for global plugins
+and plugins within a backend. It is not available to plugins opened outside
+the context of the KDB, since the global keyset is managed by the KDB instance.
```

Comment:
@markus2330 can you please take a look at this paragraph?

You made the suggestion that all plugins should get the global KS handle, which is a good idea.
The caveat is that it's hard  to do for `elektraPluginOpen`. That is why I propose we do it only for
plugins in a backend. That way the KDB passes the handle when opening a backend.

Would you agree with the solution above or did you want to think of some trickery to somehow make it work
with `elektraPluginOpen()` directly?

Reply:
Can you elaborate which plugins are affected? Is it only about the version plugin? If it is also about default resolver and default storage, this is a call for trouble.

Then this distinction seems to be arbitrary. It would be not so surprising to only pass it to global plugins. (Because of the name of the function. Non-global plugins would get NULL when they try to get the global KeySet).

Reply:
It works for the default backend and everywhere where a KDB instance exists.

Since the KDB instance "manages" the global keyset, it just can not work outside
of the context of a KDB instance. (Example: `elektraPluginOpen` can be done without
having a KDB instance available)

Reply:
I do not understand, `elektraPluginOpen` is used for all mounted plugins (see src/libs/elektra/plugin.c line 185 called from backendOpen in src/libs/elektra/backend.c).

So what exactly is "plugins opened outside the context of the KDB"? Do you mean it only affects plugins opened in test cases?

Reply:
> So what exactly is "plugins opened outside the context of the KDB"? Do you mean it only affects plugins opened in test cases?

Sorry for the misunderstanding, `elektraPluginOpen` is not the problem. The global keyset reference is set after the plugin is loaded. I already implemented it in the backends and in the list plugin to set it correctly.

Any other "custom" loading of plugins, like in the tests, does not support the global keyset.

The alternate approach is to force it by changing the `elektraPluginOpen`. I think that introduces unnecessary complexity though.

Reply:
No, you are right. elektraPluginOpen should not depend on KDB.

It is obvious that for manually instantiated plugins the global KeySet can only be available if it was passed before. Maybe rephrase the sentence so that it is clear that the "not available" only refers to manually created plugins.